moveit_core/kinematics_base/include/moveit/kinematics_base/kinematics_base.h

```diff
@@ -570,31 +580,22 @@ class KinematicsBase
   /**
    * @brief  Virtual destructor for the interface
    */
-  virtual ~KinematicsBase()
-  {
-  }
+  virtual ~KinematicsBase();
 
-  KinematicsBase()
-    : tip_frame_("DEPRECATED")
-    ,  // help users understand why this variable might not be set
-       // (if multiple tip frames provided, this variable will be unset)
-    search_discretization_(DEFAULT_SEARCH_DISCRETIZATION)
-    , default_timeout_(DEFAULT_TIMEOUT)
-  {
-    supported_methods_.push_back(DiscretizationMethods::NO_DISCRETIZATION);
-  }
+  KinematicsBase();
 
 protected:
+  moveit::core::RobotModelConstPtr robot_model_;
```

Comment:
Shouldn't we add a warning about the fake nature of this raw pointer in disguise here?

Reply:
The shared_ptr can be used as usual. There are no restrictions.